lib/acts_as_taggable_on/taggable/core.rb

```diff
@@ -107,8 +107,12 @@ def grouped_column_names_for(object)
       self.class.grouped_column_names_for(object)
     end
 
+    def unpersisted_custom_contexts
+      @unsaved_custom_contexts ||= []
+    end
+
     def custom_contexts
-      @custom_contexts ||= taggings.map(&:context).uniq
+      @custom_contexts ||= (taggings.map(&:context) + unpersisted_custom_contexts).uniq
```

Comment:
I think you have an opportunity here to do ```taggings.pluck(:context).uniq``` and it will be a query, I think it would be good to delegate to the RDBMS as much as possible. What do you think? @ProGM